sensor/common/registry/registry_store.go

```diff
@@ -392,3 +392,221 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
+}
+
+func (rs *Store) upsertSecretByHost(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	fromDefaultSA := serviceAcctName == defaultSA
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if fromDefaultSA {
+			// We assume that registries found in the dockercfg secret managed by OCP for the default
+			// service account only references hostnames for the OCP internal registry.
+			rs.addClusterLocalRegistryHost(registryAddr)
+			if err := rs.upsertRegistry(namespace, registryAddr, dce); err != nil {
+				log.Errorf("Unable to upsert registry %q into store: %v", registryAddr, err)
+			}
+			continue
+		}
+
+		if env.DelegatedScanningDisabled.BooleanSetting() {
+			// If delegated scanning is disabled then we do not store additional secrets outside of those needed
+			// for scanning images from the OCP internal registry.
+			continue
+		}
+
+		if serviceAcctName != "" {
+			// Ignore secrets for service accounts other than default so that the
+			// default registry is not overwritten in the store.
+			continue
+		}
+
+		var err error
+		if isGlobalPullSecret {
+			err = rs.upsertGlobalRegistry(registryAddr, dce)
+		} else {
+			err = rs.upsertRegistry(namespace, registryAddr, dce)
+		}
+		if err != nil {
+			log.Errorf("unable to upsert registry %q into store: %v", registryAddr, err)
+		}
+	}
+}
+
+func (rs *Store) upsertSecretByName(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	hasBoundServiceAccount := serviceAcctName != ""
+
+	// To avoid partial upserts - hold the lock until the entire secret upserted.
+	rs.storeMutux.Lock()
+	defer rs.storeMutux.Unlock()
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if hasBoundServiceAccount {
+			// We assume that registries found in dockercfg secrets managed by OCP only
+			// reference hostnames for the OCP internal registry.
+			rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
+			rs.addClusterLocalRegistryHost(registryAddr)
+			continue
+		}
+
+		if env.DelegatedScanningDisabled.BooleanSetting() {
+			// If delegated scanning is disabled then we do not store additional secrets outside of those needed
+			// for scanning images from the OCP internal registry.
+			continue
+		}
+
+		if isGlobalPullSecret {
+			if err := rs.upsertGlobalRegistry(registryAddr, dce); err != nil {
+				log.Errorf("Upserting global registry for pull secret %q, namespace %q, address %q: %v", secretName, namespace, registryAddr, err)
+			}
+		}
+
+		rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
```

Comment:
how come this isn't an `else` like it was for `upsertSecretByHost`? Can you add a comment explaining why? When reading this, it is clear these are meant to almost mirror with the addition of whatever changes between them. It'd help me know why the differences exist when coming back to this later